src/common/utils/filesystem.cc

```diff
@@ -0,0 +1,43 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+#include "src/common/utils/filesystem.h"
+
+#include <cstdio>
+
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "src/common/logging/logging.h"
+
+namespace raksha::common::utils {
+
+std::filesystem::path CreateTemporaryDirectory() {
+  std::error_code error_code;
+  std::filesystem::path temp_path =
+      std::filesystem::temp_directory_path(error_code);
+  CHECK(!error_code) << absl::StrFormat(
+      "Unable to access temporary file directory: `%s`", error_code.message());
+
+  absl::string_view temp_file = std::tmpnam(nullptr);
+  std::filesystem::path new_temp_path = temp_path / temp_file;
+  if (!std::filesystem::create_directory(new_temp_path, error_code)) {
+    LOG(FATAL) << absl::StrFormat("Unable to create directory `%s`: `%s`",
```

Comment:
Why not return `absl::StatusOr` as it used to be?  Given that this could run on an android app, we should gracefully fail. We cannot have `FATAL` failures if we can help it.

That is also why I tried to use non-exception methods from std::filesystem.

Reply:
Fixed `CreateTemporaryDirectory` to return an `absl::StatusOr` instead of failing in a `FATAL` fashion.